#! /bin/sh /usr/share/dpatch/dpatch-run
## 39_debianlvl.dpatch by Petter Reinholdtsen <pere@hungry.com>

Adjust insserv to handle 'S' in the init.d script level list as the
boot directory on debian.  A better solution would be to reengineer
insserv to have the runlevel info in one location instead of all over
the code.

@DPATCH@
--- insserv-1.08.0.orig/insserv.c
+++ insserv-1.08.0/insserv.c
@@ -418,7 +418,11 @@
 
     switch (runlvl) {
 	case 0: serv->lvls |= LVL_HALT;   break;
+#ifdef SUSE
 	case 1: serv->lvls |= LVL_ONE;    break;
+#else /* not SUSE */
+	case 1: serv->lvls |= LVL_SINGLE; break;
+#endif
 	case 2: serv->lvls |= LVL_TWO;    break;
 	case 3: serv->lvls |= LVL_THREE;  break;
 	case 4: serv->lvls |= LVL_FOUR;   break;
@@ -1133,6 +1137,7 @@
     {"../rc4.d/", LVL_FOUR,   LVL_NORM},
     {"../rc5.d/", LVL_FIVE,   LVL_NORM},
     {"../rc6.d/", LVL_REBOOT, LVL_NORM},
+    {"../rc1.d/", LVL_ONE,    LVL_NORM}, /* placeholder */
     {"../rcS.d/", LVL_BOOT,   LVL_BOOT}, /* runlevel S */
 		/* On e.g. Debian there exist no boot.d */
 #endif		/* not SUSE */
--- insserv-1.08.0.orig/listing.c
+++ insserv-1.08.0/listing.c
@@ -578,9 +578,15 @@
 	if (!strpbrk(token, "0123456sSbB"))
 	    continue;
 	if (*token == 'S' || *token == 's')
+#ifdef SUSE
 	    num = 7;
+#else
+	    num = 8;
+#endif
+#ifdef SUSE
 	else if (*token == 'B' || *token == 'b')
 	    num = 8;
+#endif /* SUSE */
 	else
 	    num = atoi(token);
 	switch (num) {
